Smt/Commands.lean

```diff
@@ -31,52 +38,42 @@ def defNatSub : Command :=
     (`"ite" • (`"<" • `"x" • `"y") • ``"0" • (`"-" • `"x" • `"y"))
     false
 
-def sortEndsWithNat : Term → Bool
-  | arrowT _ t    => sortEndsWithNat t
-  | symbolT "Nat" => true
-  | _             => false
+open ToSexp in
+protected def toSexp : Command → Sexp
+  | .setLogic l                   => sexp!{(set-logic {l})}
+  | .setOption k v                => sexp!{(set-option {s!":{k}"} {v})}
+  | .assert val                   => sexp!{(assert {toSexp val})}
```

Comment:
```suggestion
  | .assert val                   => sexp!{(assert {val})}
```
The DSL should already make the `toSexp` calls, if not then it's a bug. Similarly on lines below.

Reply:
It was only for `String`, I generalized it to `ToSexp` instances now.